src/transformers/training_args.py

```diff
@@ -207,18 +207,26 @@ class TrainingArguments:
             Random seed that will be set at the beginning of training. To ensure reproducibility across runs, use the
             :func:`~transformers.Trainer.model_init` function to instantiate the model if it has some randomly
             initialized parameters.
+        bf16 (:obj:`bool`, `optional`, defaults to :obj:`False`):
+            Whether to use bf16 16-bit (mixed) precision training instead of 32-bit training. Requires Amphere or
+            higher NVIDIA architecture.
         fp16 (:obj:`bool`, `optional`, defaults to :obj:`False`):
-            Whether to use 16-bit (mixed) precision training instead of 32-bit training.
+            Whether to use fp16 16-bit (mixed) precision training instead of 32-bit training.
         fp16_opt_level (:obj:`str`, `optional`, defaults to 'O1'):
             For :obj:`fp16` training, Apex AMP optimization level selected in ['O0', 'O1', 'O2', and 'O3']. See details
             on the `Apex documentation <https://nvidia.github.io/apex/amp.html>`__.
         fp16_backend (:obj:`str`, `optional`, defaults to :obj:`"auto"`):
+            This argument is deprecated. Use ``half_precision_backend`` instead.
+        half_precision_backend (:obj:`str`, `optional`, defaults to :obj:`"auto"`):
             The backend to use for mixed precision training. Must be one of :obj:`"auto"`, :obj:`"amp"` or
             :obj:`"apex"`. :obj:`"auto"` will use AMP or APEX depending on the PyTorch version detected, while the
             other choices will force the requested backend.
+        bf16_full_eval (:obj:`bool`, `optional`, defaults to :obj:`False`):
+            Whether to use full bfloat16 evaluation instead of 32-bit. This will be faster and save memory but can harm
+            metric values.
```

Comment:
Still not overjoyed about adding that new argument but I understand your point on this.

Reply:
I'm totally open to suggestions.

As I said the other approach is to use these 2 combinations instead:

1. `--half_precision_full_eval` + `--fp16`
2.  `--half_precision_full_eval` + `--bf16`

Reply:
> always fp16? bfloat16 when it's supported?) but it can be done in another PR.

That's not something that should be automated - as the 2 modes are very different in many ways. The user needs to make a deliberate choice.

Reply:
So let's keep the two separate flags for now, then.

Reply:
+1 on @stas00 suggestion that the data format decisions are not automated.